src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryPublisher.cs

```diff
@@ -41,6 +41,11 @@ public ValueTask PublishEvent(EvaluationEvent evaluationEvent, CancellationToken
                 properties["Variant"] = evaluationEvent.Variant.Name;
             }
 
+            if (evaluationEvent.AssignmentReason != AssignmentReason.None)
+            {
+                properties["AssignmentReason"] = evaluationEvent.AssignmentReason.ToString();
```

Comment:
Can you create a private method

```
private static string ToString(AssignemntReason reason);
```

So we don't rely on the enum name to determine our telemetry output. We wouldn't want code refactoring to introduce changes into our communication protocol.